packages/expo-camera/ios/Current/CameraView.swift

```diff
@@ -219,15 +219,28 @@ public class CameraView: ExpoView, EXCameraInterface, EXAppLifecycleListener,
     device.unlockForConfiguration()
   }
 
+  private func setIsMuted() {
+    if self.mode == .video {
+      sessionQueue.async {
+        self.session.beginConfiguration()
+        self.updateSessionAudioIsMuted()
+        self.session.commitConfiguration()
+      }
+    }
+  }
+
   private func setCameraMode() {
     sessionQueue.async {
+      self.session.beginConfiguration()
       if self.mode == .video {
         if self.videoFileOutput == nil {
           self.setupMovieFileCapture()
         }
+        self.updateSessionAudioIsMuted()
```

Comment:
I think this is the only change the is necessary to fix the mute issue. Can you confirm that @lodev09? If we still need to deal with an artificial delay for your use case then the other changes are not solving the problem

Reply:
The reason for this is I'm very cautious about moving the configuration calls. We can easily get into a situation that causes crashes that is not immediately obvious during testing

Reply:
We can look at another solution for the dead frames problem when immediately recording after a switch

Reply:
@alanjhughes I will check. As I mentioned before, adding only that in that block adds longer dead frame and I'm suspecting it's because of the session queue unit. I can try to revert back the configuration calls.